InventoryInStorePickupGraphQl/Model/Resolver/PickupLocations/SearchRequest/ProductsInfo.php

```diff
@@ -0,0 +1,81 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\InventoryInStorePickupGraphQl\Model\Resolver\PickupLocations\SearchRequest;
+
+use Magento\InventoryInStorePickupApi\Api\Data\SearchRequest\ProductInfoInterfaceFactory;
+use Magento\InventoryInStorePickupApi\Api\Data\SearchRequestExtensionFactory;
+use Magento\InventoryInStorePickupApi\Api\Data\SearchRequestExtensionInterface;
+use Magento\InventoryInStorePickupApi\Model\SearchRequestBuilderInterface;
+
+/**
+ * Resolve Products Information.
+ */
+class ProductsInfo implements ResolverInterface
+{
+    private const SKU = 'sku';
+
+    /**
+     * @var SearchRequestExtensionFactory
+     */
+    private $searchRequestExtensionFactory;
+
+    /**
+     * @var ProductInfoInterfaceFactory
+     */
+    private $productInfoInterfaceFactory;
+
+    public function __construct(
+        SearchRequestExtensionFactory $searchRequestExtensionFactory,
+        ProductInfoInterfaceFactory $productInfoInterfaceFactory
+    ) {
+        $this->searchRequestExtensionFactory = $searchRequestExtensionFactory;
+        $this->productInfoInterfaceFactory = $productInfoInterfaceFactory;
+    }
+
+    /**
+     * @@inheritdoc
+     */
+    public function resolve(
+        SearchRequestBuilderInterface $searchRequestBuilder,
+        string $fieldName,
+        string $argumentName,
+        array $argument
+    ): SearchRequestBuilderInterface {
+        $productsInfo = [];
+        foreach ($argument[$argumentName] as $productInfoData) {
+            $productsInfo[] = $this->productInfoInterfaceFactory->create([self::SKU => $productInfoData[self::SKU]]);
+        }
+        $extension = $this->getExtensionAttributes($searchRequestBuilder);
+
+        $extension->setProductsInfo($productsInfo);
+        $searchRequestBuilder->setSearchRequestExtension($extension);
+
+        return $searchRequestBuilder;
+    }
+
+    /**
+     * Get Search Request Extension Attributes DTO.
+     *
+     * @param SearchRequestBuilderInterface $searchRequestBuilder
+     *
+     * @return SearchRequestExtensionInterface
+     */
+    private function getExtensionAttributes(
+        SearchRequestBuilderInterface $searchRequestBuilder
+    ): SearchRequestExtensionInterface {
+        $builderCopy = clone $searchRequestBuilder;
```

Comment:
There is a problem that we can override extension attributes which were previously added to the builder.
I see 3 possible solutions for the problem:
1) Current implementation - build copy of search request and get extension attributes from it.
2) Ignore the problem. Could cause a problem for extension developers and solution providers - only one class will be allowed to set extension attributes.
3) Rethink service contracts. Probably, add new interface which will have 'getExtensionAttributes' or 'getData' for the builder.
Any suggestions?

Reply:
I think we must follow option 3, but still have no idea what the best solution is.